### ISSUE TITLE
"Failed to execute w3mimgdisplay, []" but only on JPEGs

### ISSUE THREAD

**Issue by b33rcity** [ranger/ranger#1170]
#### Runtime Environment

ranger-stable 1.8.1
Python 3.5.5 (default, May 13 2018, 20:10:32) 
[GCC 6.4.0]
Termite, Gentoo, awesomwm

#### Current Behavior
I can preview png files in Termite with the w3m method without a problem. JPEG files consistently return the error in the title and refuse to display. This also affects PDF files, regardless of whether I use pdftoppm or pdftotext. This issue was present with my config files from the ebuild I installed on 2018-04-08 and persisted after I updated the ebuilt and re-copied the config files. 

#### Expected Behavior
JPEG (and PDF) should display as PNGs do. 

#### Steps to reproduce
1. Install ebuild from Portage of: Ranger, Python3.5, and w3m; compile Termite and vte-ng from github source
2. Set preview_images=true and method to w3m
3. Attempt to view PNG and JPEG images

#### Traceback

```
Failed to execute w3mimgdisplay, []
```

**Comment by toonn** [comment]
Does w3migmdisplay work for you from the command line?

**Comment by b33rcity** [comment]
Yes, I can call `w3m /path/to/image.jpg` and it will pop up a new window showing the image just fine. 

Tbh I'm fuzzy on where one thing ends and the other begins--I think calling w3m is opening up ImageMagick in a new window, and I'm really not sure if this is what Ranger tries to do or not.

**Comment by toonn** [comment]
It's probably termite. w3mimgdisplay is made with xterm in mind and termite's too different.

**Comment by b33rcity** [comment]
I installed an ebuild of xterm just now and found the same issue. Ranger shows the png but not the jpg. I'll research w3mimgdisplay a bit more. It's possible I missed something in the w3m ebuild (I have all the USE flags listed in the Portage docs for w3m, but you never know with Gentoo).

**Comment by b33rcity** [comment]
Quick update:

I tried installing w3m from source on Arch and termite from pacman. Image previews using w3m worked flawlessly following this procedure. 

Optimistic, I tried unmerging w3m on my Gentoo box and reinstalling it from source only to run into the exact same issue. This may be a peculiarity of Gentoo (big fricken' surprise).

**Comment by b33rcity** [comment]
Thanks to some help on the Gentoo Forums, I was able to find a solution to this problem. 

The issue wasn't with ranger or even with w3m, but with a dependency of w3m. In my case, imlib2 was not compiled with jpeg support. I'm not sure how that happened, but once I found the problem it was a simple matter of running `emerge -N` on that package.

Termite is now happily playing with w3m in Ranger--and it's even transparent and supports Powerline fonts!
